onmt/tests/test_beam_search.py

```diff
@@ -526,11 +516,10 @@ def third_step(self, beam, expected_beam_scores, expected_len_pen):
             expected_beam_scores / expected_len_pen))
         self.assertTrue(beam.topk_ids.equal(expected_preds_3))
         self.assertTrue(beam.current_backptr.equal(expected_bptr_3))
-        self.assertEqual(beam.is_finished.sum(), self.BATCH_SZ)
-        # new beam 1 finished
-        self.assertTrue(beam.is_finished[:, 1].all())
-        # new beam 1 is old beam 4
-        self.assertTrue(expected_bptr_3[:, 1].eq(4).all())
+        # we finish 3 hyps per example in this step
```

Comment:
Why were the test changed ?

Reply:
As stated above we no longer test the same thing given that 3 beams finish instead of 1 in this step

Reply:
what about self.assertTrue(beam.is_finished[:, 1].all()) ?

Reply:
but then we can also add:
self.assertTrue(beam.is_finished[:, 0].all())
self.assertTrue(beam.is_finished[:, 2].all())
to me the most important thing is to make sure beam.done is true, specifying the 3 that died is not relevant, but if you insist then we need to add the 3 of them.